test/jackdaw/streams_test.clj

```diff
@@ -443,6 +444,101 @@
         (is (= [2 3] (second keyvals)))
         (is (= [2 7] (nth keyvals 2))))))
 
+ (testing "flat-transform"
+    (let [topic-a (mock/topic "topic-a")
+          topic-b (mock/topic "topic-b")
+          transformer-supplier-fn #(let [total (atom 0)]
+                                    (reify Transformer
+                                      (init [_ _])
+                                      (close [_])
+                                      (transform [_ k v]
+                                        ;; each input creates two outputs
+                                        ;; each v' accumulating the v read in:
+                                        ;; [[k * 10, v'] [k * 20, v'']]
+                                        (map (fn [x]
+                                               (swap! total + v)
+                                               (key-value [(* k x) @total]))
+                                             [10 20]))))
+          driver (mock/build-driver (fn [builder]
+                                      (-> builder
+                                          (k/kstream topic-a)
+                                          (k/flat-transform transformer-supplier-fn)
+                                          (k/to topic-b))))
+          publish (partial mock/publish driver topic-a)]
+
+      (publish 1 1)
+      (publish 1 2)
+      (publish 1 4)
+
+      (let [keyvals (mock/get-keyvals driver topic-b)]
+        (is (= 6 (count keyvals)))
+        (is (= [10 1] (first keyvals)))
+        (is (= [20 2] (second keyvals)))
+        (is (= [10 4] (nth keyvals 2)))
+        (is (= [20 6] (nth keyvals 3)))
+        (is (= [10 10] (nth keyvals 4)))
+        (is (= [20 14] (nth keyvals 5))))))
+
+  (testing "with-state-store and transform-with-ctx"
+    (let [topic-a (mock/topic "topic-a")
+          topic-b (mock/topic "topic-b")
+          driver (mock/build-driver (fn [builder]
+                                      (-> builder
+                                          (k/with-state-store {:store-name "test-store"})
+                                          (k/kstream topic-a)
+                                          (k/transform
+                                           (lambdas/transformer-with-ctx
```

Comment:
Wondering if a macro could make this feel more 'clojurey' like 
```clojure
(transformer-with-context [test-store k v]
(update test-store k some-fn)
)
```

Reply:
Yeah there is possible a macro here as the state store names have to be repeated. I try to avoid reaching for macro's immediately however until the basic functions are there.

We could perhaps start a macros namespace, as there are probably several syntax things you could add over the basic functions for use in streams.